### PR TITLE
[Documentation][Cookbook] How to add a custom promotion rule?

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| BC breaks? | no |
| Related tickets | partially #5275 |
| License | MIT |

### CODE REVIEW

File: docs/cookbook/custom-promotion-rule.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    class PremiumCustomerConfigurationType extends AbstractType
+    {
+        /**
+         * {@inheritdoc}
+         */
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('amount', 'sylius_money', [
+                    'label' => 'app.form.promotion_rule.premium_customer_configuration.amount',
```
Comment (michalmarcinkowski) [inline docs/cookbook/custom-promotion-rule.rst]: Do we need `amount` in this configuration? We do not use this value anywhere. IMO this rule shouldn't have any configuration fields.

Comment (TheMadeleine) [inline docs/cookbook/custom-promotion-rule.rst]: That's true. :) fixed.

File: docs/cookbook/custom-promotion-rule.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\Promotion\Checker\Rule;
+
+    use Sylius\Component\Promotion\Checker\Rule\RuleCheckerInterface;
+    use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+
+
```
Comment (pamil) [inline docs/cookbook/custom-promotion-rule.rst]: Two blank lines in a row :)

File: docs/cookbook/custom-promotion-rule.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    class PremiumCustomerRuleChecker implements RuleCheckerInterface
+    {
+        const TYPE = 'premium_customer';
```
Comment (pamil) [inline docs/cookbook/custom-promotion-rule.rst]: Where is it used? Why is it needed?

Comment (TheMadeleine) [inline docs/cookbook/custom-promotion-rule.rst]: Each rule checker has this constant to be able to create a new rule with type programmatically :)
```
$rule->setType(PremiumCustomerRuleChecker::TYPE);
```

**Comment by pjedrzejewski** [comment]
Thanks Magda! :)

**Pull request merged**